forester/src/photon_indexer.rs

```diff
@@ -178,4 +186,69 @@ impl<R: RpcConnection> Indexer<R> for PhotonIndexer<R> {
 
         Ok(proofs)
     }
+
+    async fn get_compressed_accounts_by_owner(
+        &self,
+        owner: &Pubkey,
+    ) -> Vec<CompressedAccountWithMerkleContext> {
+        let params = GetCompressedAccountsByOwnerPostRequestParams {
+            cursor: None,
+            limit: None,
+            owner: owner.to_string(),
+            ..Default::default()
+        };
+        let request = photon_api::models::GetCompressedAccountsByOwnerPostRequest {
+            params: Box::from(params),
+            ..Default::default()
+        };
+
+        let result = photon_api::apis::default_api::get_compressed_accounts_by_owner_post(
+            &self.configuration,
+            request,
+        )
+        .await;
+        // let env_accounts = EnvAccounts::get_local_test_validator_accounts();
+        match result {
+            Ok(response) => match response.result {
+                Some(result) => {
+                    let accounts = result
+                        .value
+                        .items
+                        .iter()
+                        .map(|x| {
+                            let address = x.address.as_ref().map(|address| decode_hash(address));
+                            let data = x.data.as_ref().map(|data| CompressedAccountData {
+                                data: bs58::decode(data.data.clone()).into_vec().unwrap(),
+                                discriminator: (data.discriminator as u64).to_le_bytes(),
+                                data_hash: decode_hash(&data.data_hash),
+                            });
+                            use std::str::FromStr;
+                            let compressed_account = CompressedAccount {
+                                address,
+                                owner: Pubkey::from_str(x.owner.as_str()).unwrap(),
+                                data,
+                                lamports: x.lamports.try_into().unwrap(),
+                            };
+                            let merkle_context = MerkleContext {
+                                merkle_tree_pubkey: Pubkey::new_from_array(decode_hash(&x.tree)),
+                                leaf_index: x.leaf_index.try_into().unwrap(),
+                                queue_index: None,
+                                nullifier_queue_pubkey: Pubkey::default(), //env_accounts.nullifier_queue_pubkey, // TODO: make dynamic. Why don't we get this value from the rpc call?
```

Comment:
I think the correct approach here would be fetching the MT account and checking the associated queue. We could consider caching the results in a local hashmap. Do you agree? That's what I'm trying right now on my side.